app/models/global_setting.rb

```diff
@@ -85,6 +85,10 @@ def self.use_s3?
       end) == :true
   end
 
+  def self.s3_bucket_name
+    @s3_bucket_name ||= s3_bucket.downcase.split("/")[0]
```

Comment:
If the `s3_bucket` contains more than just the bucket name, do we consider that a misconfiguration?

Reply:
Commit aa5de3c40acebe3f3577a977cf57ed7630a4b740 added support for subfolders in the bucket name as "bucket/folder". And that works fine if you put that in the admin section.

But if you pass it in via environment variable (which is how it gets stuffed into discourse.conf) then the absolute_base_url is wrong. This PR fixes that.

Having a separate config var for the subfolder is probably a good idea? But that's a bigger change.